nmostesting/mocks/Auth.py

```diff
@@ -0,0 +1,535 @@
+        self.code = None
+
+    def make_mdns_info(self, priority=0, api_ver=None, ip=None):
+        """Get an mDNS ServiceInfo object in order to create an advertisement"""
+        if api_ver is None:
+            api_ver = self.version
+
+        api_proto = self.protocol
+
+        if ip is None:
+            ip = get_default_ip()
+            hostname = "nmos-mocks.local."
+        else:
+            hostname = ip.replace(".", "-") + ".local."
+
+        port = self.port
+
+        # TODO: Add another test which checks support for parsing CSV string in api_ver
+        txt = {'api_ver': api_ver, 'api_proto': api_proto, 'pri': str(priority)}
+
+        service_type = "_nmos-auth._tcp.local."
+        info = ServiceInfo(service_type,
+                           "NMOSTestSuite{}{}.{}".format(port, api_proto, service_type),
+                           addresses=[socket.inet_aton(ip)], port=port,
+                           properties=txt, server=hostname)
+        return info
+
+    def make_issuer(self):
+        return "{}://{}:{}".format(self.protocol, self.host, self.port)
+
+    def make_metadata(self):
+        metadata = {
+            "issuer": self.make_issuer(),
+            "authorization_endpoint": "{}://{}:{}/auth".format(self.protocol, self.host, self.port),
+            "token_endpoint": "{}://{}:{}/token".format(self.protocol, self.host, self.port),
+            "jwks_uri": "{}://{}:{}/jwks".format(self.protocol, self.host, self.port),
+            "registration_endpoint": "{}://{}:{}/register".format(self.protocol, self.host, self.port),
+            "response_types_supported": [
+                "code",
+                "code token"
+            ]
+        }
+        metadata["grant_types_supported"] = [e.name for e in GRANT_TYPES]
+        metadata["scopes_supported"] = [e.name for e in SCOPES]
+        metadata["code_challenge_methods_supported"] = [e.name for e in CODE_CHALLENGE_METHODS]
+        return metadata
+
+    def generate_jwk(self):
+        private_key = IS10Utils.read_RSA_private_key(self.private_keys)
+        return IS10Utils.generate_jwk(private_key)
+
+    def generate_token(self, scopes=None, write=False, azp=False, add_claims=True, exp=3600, overrides=None):
+        private_key = IS10Utils.read_RSA_private_key(self.private_keys)
+        overrides_ = {"iss": "{}://{}:{}".format(self.protocol, self.host, self.port)}
+        if overrides:
+            overrides_.update(overrides)
+        return IS10Utils.generate_token(private_key, scopes, write, azp, add_claims, exp=exp, overrides=overrides_)
+
+
+# 0 = Primary mock Authorization API
+# 1 = Secondary mock Authorization API, which is used for testing resources server on handling
+#     access token from different Authorization server
+NUM_AUTHS = 2
+AUTHS = [Auth(i + 1) for i in range(NUM_AUTHS)]
+AUTH_API = Blueprint('auth_pi', __name__)
+PRIMARY_AUTH = AUTHS[0]
+SECONDARY_AUTH = AUTHS[1]
+
+SPEC_PATH = os.path.join("cache/is-10")
```

Comment:
This should be using `CONFIG.CACHE_PATH`?

https://github.com/AMWA-TV/nmos-testing/blob/5b3f1eca4db1f37d8828e37eae8e93f108529946/nmostesting/Config.py#L65-L66